universal-login-contracts/contracts/ProxyCounterfactualFactory.sol

```diff
@@ -0,0 +1,33 @@
+pragma solidity ^0.5.2;
+
+import "openzeppelin-solidity/contracts/ownership/Ownable.sol";
+
+
+contract ProxyCounterfactualFactory is Ownable {
+    address public contractAddress;
+
+    function createContract(address signer, address publicKey, bytes memory initCode, bytes memory initializeWithENS) public onlyOwner returns(bool success) {
+        bytes32 finalSalt = keccak256(abi.encodePacked(publicKey, signer));
```

Comment:
Since we use the only signer (relayer) that is the owner of the contract, the contract address already is depended on private/publicKey of a relayer. No need to include `signer` in hash.